cpp/src/arrow/compute/exec/asof_join_benchmark.cc

```diff
@@ -49,6 +49,20 @@ static Result<TableStats> MakeTable(const TableGenerationProperties& properties)
   return Result<TableStats>({table, rows, rows * row_size});
 }
 
+// As opposed to using table_source, we create a make a ReaderGenerator for the specified
+// table. This allows us to specify a thread pool to isolate the threads used for each
+// source as an anti-deadlocking mechanism.
+static ExecNode* MakeTableSourceNode(std::shared_ptr<arrow::compute::ExecPlan> plan,
```

Comment:
This is a minor nitpick but you can probably just pass `ExecPlan*` here since that is all `MakeExecNode` requires.  When you pass `std::shared_ptr` you are either forcing a copy of the `shared_ptr` (unless the caller can `std::move` which I don't think is the case here).  In general that isn't a big deal (one atomic increment) but it's good to get in the practice of only using what you need.

Reply:
Change to `ExecPlan*`!